pennylane/qchem/convert.py

```diff
@@ -1202,4 +1185,85 @@ def _shci_state(wavefunction, tol=1e-15):
     # filter based on tolerance cutoff
     dict_fcimatr = {key: value for key, value in dict_fcimatr.items() if abs(value) > tol}
 
+    # convert sign parity from chemist to physicist convention (interleaving spin operators
+    # rather than commuting all spin-up operators to the left)
+    dict_fcimatr = _sign_chem_to_phys(dict_fcimatr, norb)
+
     return dict_fcimatr
+
+
+def _sitevec_to_fock(det, format):
+    r"""Convert a Slater determinant from site vector to occupation number vector representation.
+
+    Args:
+        det (list): determinant in site vector representation
+        format (str): the format of the determinant
+
+    Returns:
+        tuple: tuple of integers representing binaries that correspond to occupation vectors in
+            alpha and beta spin sectors
+
+    **Example**
+
+    >>> det = [1, 2, 1, 0, 0, 2]
+    >>> _sitevec_to_fock(det, format = 'dmrg')
+    (5, 34)
+
+    >>> det = ["a", "b", "a", "0", "0", "b"]
+    >>> _sitevec_to_fock(det, format = 'shci')
+    (5, 34)
+    """
+
+    if format == "dmrg":
+        format_map = {0: "00", 1: "10", 2: "01", 3: "11"}
+    elif format == "shci":
+        format_map = {"0": "00", "a": "10", "b": "01", "2": "11"}
+
+    strab = [format_map[key] for key in det]
+
+    stra = "".join(i[0] for i in strab)
+    strb = "".join(i[1] for i in strab)
+
+    inta = int(stra[::-1], 2)
+    intb = int(strb[::-1], 2)
+
+    return inta, intb
+
+
+def _sign_chem_to_phys(dict_fcimatr, norb):
+    r"""Convert the dictionary-form wavefunction from chemist sign convention
+    for ordering the creation operators by spin (i.e. all spin-up operators
+    on the left) to the physicist convention native to PennyLane, which
+    storing spin operators as interleaved for the same spatial orbital index.
+
+    Note that convention change in the opposite direction -- starting from physicist
+    and going to chemist -- can be accomplished with the same function
+    (the sign transformation is reversible).
+
+    Args:
+        dict_fcimatr (dict): dictionary of the form `{(int_a, int_b) :coeff}`, with integers `int_a, int_b`
```

Comment:
```suggestion
        dict_fcimatr (dict[tuple(int), float]): dictionary of the form `{(int_a, int_b) :coeff}`, with integers `int_a, int_b`
```
It would be good to add type hinting in the docs like this as well, in addition to the explanation given in the docs. Also, you might prefer using `fcimatr_dict` instead of `dict_fcimatr`.